contrib/babelfishpg_common/sql/upgrades/babelfishpg_common--2.5.0--2.6.0.sql

```diff
@@ -1,12 +1,147 @@
+WITH FUNCTION sys.binarysysvarchar (sys.BBF_BINARY, integer, boolean) AS ASSIGNMENT;
+
+CALL sys.babelfish_drop_deprecated_object('function', 'sys', 'binarysysvarchar_deprecated_2_6_0');
+
+-- (sys.BBF_BINARY AS pg_catalog.VARCHAR)
+DROP CAST IF EXISTS(sys.BBF_BINARY AS pg_catalog.VARCHAR);
+
+DO $$
+DECLARE
+    exception_message text;
+BEGIN
+    ALTER FUNCTION sys.binaryvarchar(sys.BBF_BINARY) RENAME TO binaryvarchar_deprecated_2_6_0;
+
+EXCEPTION WHEN OTHERS THEN
+    GET STACKED DIAGNOSTICS
+    exception_message = MESSAGE_TEXT;
+    RAISE WARNING '%', exception_message;
+END;
+$$;
+
+CREATE OR REPLACE FUNCTION sys.binaryvarchar(sys.BBF_BINARY, integer, boolean)
+RETURNS pg_catalog.VARCHAR
+AS 'babelfishpg_common', 'varbinaryvarchar'
+LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+CREATE CAST (sys.BBF_BINARY AS pg_catalog.VARCHAR)
+WITH FUNCTION sys.binaryvarchar (sys.BBF_BINARY, integer, boolean) AS ASSIGNMENT;
+
+CALL sys.babelfish_drop_deprecated_object('function', 'sys', 'binaryvarchar_deprecated_2_6_0');
+
+-- (sys.ROWVERSION AS sys.VARCHAR)
+DROP CAST IF EXISTS(sys.ROWVERSION AS sys.VARCHAR);
+
+DO $$
+DECLARE
+    exception_message text;
+BEGIN
+    ALTER FUNCTION sys.rowversionsysvarchar(sys.ROWVERSION) RENAME TO rowversionsysvarchar_deprecated_2_6_0;
+
+EXCEPTION WHEN OTHERS THEN
+    GET STACKED DIAGNOSTICS
+    exception_message = MESSAGE_TEXT;
+    RAISE WARNING '%', exception_message;
+END;
+$$;
+
+CREATE OR REPLACE FUNCTION sys.rowversionsysvarchar(sys.ROWVERSION, integer, boolean)
+RETURNS sys.VARCHAR
+AS 'babelfishpg_common', 'varbinaryvarchar'
+LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+CREATE CAST (sys.ROWVERSION AS sys.VARCHAR)
+WITH FUNCTION sys.rowversionsysvarchar (sys.ROWVERSION, integer, boolean) AS ASSIGNMENT;
+
+CALL sys.babelfish_drop_deprecated_object('function', 'sys', 'rowversionsysvarchar_deprecated_2_6_0');
+
+-- (sys.BBF_BINARY AS pg_catalog.VARCHAR)
+DROP CAST IF EXISTS(sys.ROWVERSION AS pg_catalog.VARCHAR);
+
+DO $$
+DECLARE
+    exception_message text;
+BEGIN
+    ALTER FUNCTION sys.rowversionvarchar(sys.ROWVERSION) RENAME TO rowversionvarchar_deprecated_2_6_0;
+
+EXCEPTION WHEN OTHERS THEN
+    GET STACKED DIAGNOSTICS
+    exception_message = MESSAGE_TEXT;
+    RAISE WARNING '%', exception_message;
+END;
+$$;
+
+CREATE OR REPLACE FUNCTION sys.rowversionvarchar(sys.ROWVERSION, integer, boolean)
+RETURNS pg_catalog.VARCHAR
+AS 'babelfishpg_common', 'varbinaryvarchar'
+LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+CREATE CAST (sys.ROWVERSION AS pg_catalog.VARCHAR)
+WITH FUNCTION sys.rowversionvarchar (sys.ROWVERSION, integer, boolean) AS ASSIGNMENT;
+
+CALL sys.babelfish_drop_deprecated_object('function', 'sys', 'rowversionvarchar_deprecated_2_6_0');
+
+-- Drops the temporary procedure used by the upgrade script.
+-- Please have this be one of the last statements executed in this upgrade script.
+DROP PROCEDURE sys.babelfish_drop_deprecated_object(varchar, varchar, varchar);
```

Comment:
Have we verified all the possible data type cast to varchar? apart from [var]binary and rowversion?

Reply:
Yes. The tests for other possible datatype cast to varchar already exists in babel_datatype.sql